### PR TITLE
Fix empty result issue and scanner closing issue

### PR DESCRIPTION
**Pull request by gfanfp101**
1. fix an issue of returning empty result when filterCallback is null
2. cherry-pick a commit from OpenTSDB to fix closing scanner issue due to hbase version incompatibility 

@jasonculverhouse @lzheng

### CODE REVIEW

**Comment by eden** [comment]
Should we just bring this up to OpenTSDB master?

**Comment by gfanfp101** [comment]
I tried to merge OpenTSDB master, there are many divergencies shown during their progress of the master, as well as we have many customized changes in our version. Just realized it is a lot of work in fixing conflict and testing for both old and new hbase client in our platform. So this PR is a quick fix of my need of hbase scanner. I tested flirt server on cl6-a (0.98) for the scanner, it works well. I think @lzheng and I need to run this on UrlContentTable to make sure it works for hbase-d (0.94).

File: src/Scanner.java [hunk 1]
```diff
@@ -1492,6 +1492,7 @@ ChannelBuffer serialize(final byte server_version) {
       final ScanRequest req = ScanRequest.newBuilder()
         .setScannerId(scanner_id)
         .setCloseScanner(true)
+        .setNumberOfRows(0)
```
Comment (lzheng) [inline src/Scanner.java]: Is this necessary? I assume it should use a default value here?

**Comment by lzheng** [comment]
+1

@eden Jason and I chatted. We thought we should just get rid of observer so we can directly use asynchbase. That'll save us more troubles in the future.

File: src/Scanner.java (hunk 1 above)
Comment (gfanfp101) [inline src/Scanner.java]: this is specifically for closeScannerRequest, which expect 0 row as a response. not hundred percent sure the internal change, but based on the bug fix log of hbaseAsync, the format changed some where between 0.94 to 0.98. It does solved my exception.

**Comment by gfanfp101** [comment]
I agree you guys, I expect that would take some more work. Can we merge this as a temporary solution of scanner closing issue ?

**Comment by lzheng** [comment]
sounds good to me.

**Pull request merged**